### PR TITLE
DB locking

### PR DESCRIPTION
**Pull request by NatasaKralj**
Addresses issue https://github.com/mendix/docs/issues/5198.
This PR adds more information on how database records get locked and how transactions are isolated.

### CODE REVIEW

**Comment by NatasaKralj** [comment]
Added updates after technical review by dbstorage team.

File: content/en/docs/developerportal/operate/logs.md [hunk 1]
```diff
@@ -87,7 +87,11 @@ The Kibana user guide is available here: [Kibana User Guide](https://www.elastic
 
 ## 5 Free App
 
-Logs are not available for Free Apps. You will see a sample screen.
+Logs are not available for Free Apps. You will see a sample screen on the logs page. You only have access to a live log of your deployed free app, on the **Environments** page, under **View Live Log**. 
+
+{{% alert color="info" %}}
+If you have set up custom log messages, you will only be able to see messages up to Info log level.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/operate/logs.md]: Thanks for adding that - it was on my to do list too!
Perhaps, to be absolutely clear:
```suggestion
If you have set up custom log messages, you will only be able to see messages up to Info log level. Debug and Trace log messages will not be shown in the live log.
```

File: content/en/docs/refguide/runtime/data-storage/_index.md [hunk 2]
```diff
@@ -55,3 +55,19 @@ You now try to deploy `Order Viewer` to use the same database as `Order Processi
...
 
 An alternative is to copy the data to another app, for example using the [Database Replication](/appstore/modules/database-replication/) module. This however, will be a snapshot of your data at the time you replicate it and changes to the data made in the original app will not be reflected in your new app.
+
+## 4 Database Transactions and Locking
+
+### 4.1 Database Record Locks
+
+Mendix does not use read locks on the database. Therefore, object reads can always be performed. A database write lock is put in place for the first write operation on that object. If there are overlapping update transactions on the same object, the first update transaction puts a write lock on the database object. The next update transaction waits until the first one finishes before executing. However, if the first update process takes an inordinate amount of time, it may cause a timeout.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/data-storage/_index.md]: Didn't we find that the write lock is only put in place when you do a commit? Or was I misreading the sequence of messages?

File: content/en/docs/refguide/runtime/data-storage/_index.md [hunk 3]
```diff
@@ -55,3 +55,19 @@ You now try to deploy `Order Viewer` to use the same database as `Order Processi
...
+
+When users or microflows make changes to database records, all changes will execute in a transaction. Write locks are placed on individual records the moment they are changed. Until this point no locks will be placed on the data. For more information on the differences between transaction commits and object commits, see the [How Commits Work](/refguide/committing-objects/#how-commits-work) section in *Commit Object(s)*.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/data-storage/_index.md]: the moment they are committed?

File: content/en/docs/refguide/runtime/data-storage/_index.md [hunk 4]
```diff
@@ -55,3 +55,19 @@ You now try to deploy `Order Viewer` to use the same database as `Order Processi
...
+
+When the record gets locked, as long as the transaction is executing, no other users or processes are able to change the data. The uncommitted information is not visible for others. The changed data becomes available for other users to read only after the transaction completes. While the transaction is running, other users are able to read and change the previously persisted version of the data. Any changes from other processes will be applied when the transaction completes and the initial write lock is lifted, overwriting the previous changes.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/data-storage/_index.md:67]: Yes, I like this.

**Review by MarkvanMents — COMMENTED** [review]
Great change.
I have made a comment about "writing" the change and "committing" the change. I _think_ we noticed that the record was only locked when you did a commit - but I haven't got the sequence of log messages in front of me so I can't be sure.

But once this is confirmed by R&D it will be a great improvement.

File: content/en/docs/refguide/runtime/data-storage/_index.md (hunk 2 above)
Comment (basdebakker) [reply content/en/docs/refguide/runtime/data-storage/_index.md]: That is true. When I say "write" I mean "object commit" as this causes the write to the database, but I can imagine a user thinking of "change object action".

File: content/en/docs/refguide/runtime/data-storage/_index.md (hunk 3 above)
Comment (NatasaKralj) [reply content/en/docs/refguide/runtime/data-storage/_index.md]: Changed

File: content/en/docs/refguide/runtime/data-storage/_index.md (hunk 2 above)
Comment (NatasaKralj) [reply content/en/docs/refguide/runtime/data-storage/_index.md]: Changed "write operation on that object" to "object commit".

**Comment by MarkvanMents** [comment]
Approved over Slack

**Pull request merged**